src/Tasks.UnitTests/MSBuild_Tests.cs

```diff
@@ -1408,5 +1407,76 @@ public void NonexistentTargetFailsAfterSkipped()
             // 2nd invocation of t_missing should fail the build resulting in target not found error (MSB4057)
             logger.FullLog.ShouldContain("MSB4057");
         }
+
+        [Fact]
+        public void MSBuildTaskPassesTaskIdToSpawnedBuilds()
+        {
+            string projectFile1 = ObjectModelHelpers.CreateTempFileOnDisk(@"
+                <Project>
+                    <Target Name=`Build`>
+                        <Message Text=`test`/>
+                    </Target>
+                </Project>");
+
+            string projectFile2 = ObjectModelHelpers.CreateTempFileOnDisk(@"
+                <Project>
+                    <Target Name=`Build`>
+                        <MSBuild Projects=`" + projectFile1 + @"` Targets=`Build` />
+                    </Target>	
+                </Project>");
+
+            try
+            {
+                Project project = new Project(projectFile2);
+                var logger = new MSBuildTaskIdLogger();
+
+                Assert.True(project.Build(logger));
+
+                var text = logger.Text;
+                var taskId = Regex.Match(text, @"TaskId=(\d+)").Groups[1].Value;
+                var parentTaskId = Regex.Match(text, @"ParentTaskId=(\d+)").Groups[1].Value;
+                Assert.Equal(taskId, parentTaskId);
+            }
+            finally
+            {
+                File.Delete(projectFile1);
```

Comment:
You can use the fancy [TestEnvironment](https://github.com/microsoft/msbuild/blob/master/src/Shared/UnitTests/TestEnvironment.cs) to avoid manual cleanup.

Reply:
I followed the pattern in the file :) We can convert to TestEnvironment later in a separate change. Might as well do it for all other tests then.